modules/api/src/main/java/com/intuit/wasabi/api/ApplicationsResource.java

```diff
@@ -395,19 +396,13 @@ public Response getAllExperimentTags(
             @ApiParam(value = EXAMPLE_AUTHORIZATION_HEADER, required = true)
             final String authorizationHeader) {
         try {
-
-            // get the for the user visible Applications
-            List<Application.Name> allApplications = experiments.getApplications();
-            Set<Application.Name> allowed = new HashSet<>();
             UserInfo.Username userName = authorization.getUser(authorizationHeader);
+            Set<Application.Name> allowed = new HashSet<>();
 
-            for (Application.Name applicationName : allApplications) {
-                try {
-                    authorization.checkUserPermissions(userName, applicationName, READ);
-                    allowed.add(applicationName);
-                } catch (AuthenticationException ignored) {
-                    LOGGER.trace("ignoring authentication exception", ignored);
-                }
+            // get the for the user visible Applications
+            List<UserPermissions> authorized = authorization.getUserPermissionsList(userName).getPermissionsList();
+            for (UserPermissions perm : authorized) {
+                allowed.add(perm.getApplicationName());
```

Comment:
@AndreaSuckro Just a question here! I saw that we were explicitly checking for `READ` permission in your old code but we do not check for the permission type now. It makes sense though to not check the permission type as I presume `READ` would be the lowest permission type and if there is a record for the user in the permissions table with the given app, we can safely consider the user authorized for atleast `READ` permissions. Correct?